host/containerinit/init.go

```diff
@@ -349,37 +347,37 @@ func setupNetworking(args *ContainerInitArgs) error {
 	return nil
 }
 
-func getCredential(args *ContainerInitArgs) (*syscall.Credential, error) {
-	if args.user == "" {
+func getCredential(c *Config) (*syscall.Credential, error) {
+	if c.User == "" {
 		return nil, nil
 	}
 	users, err := user.ParsePasswdFileFilter("/etc/passwd", func(u user.User) bool {
-		return u.Name == args.user
+		return u.Name == c.User
 	})
 	if err != nil || len(users) == 0 {
 		if err == nil {
 			err = errors.New("unknown user")
 		}
-		return nil, fmt.Errorf("Unable to find user %v: %v", args.user, err)
+		return nil, fmt.Errorf("Unable to find user %v: %v", c.User, err)
 	}
 
 	return &syscall.Credential{Uid: uint32(users[0].Uid), Gid: uint32(users[0].Gid)}, nil
 }
 
-func setupCommon(args *ContainerInitArgs) error {
-	if err := setupHostname(args); err != nil {
+func setupCommon(c *Config) error {
+	if err := setupHostname(c); err != nil {
 		return err
 	}
 
-	if err := setupNetworking(args); err != nil {
+	if err := setupNetworking(c); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func getEnv(args *ContainerInitArgs, key string) string {
-	for _, kv := range args.env {
+func getEnv(c *Config, key string) string {
```

Comment:
We need it to be a `[]string` when we pass it to `os/exec` so merging here is fine.